neural_compressor/adaptor/ox_utils/smooth_quant.py

```diff
@@ -0,0 +1,594 @@
+                                    }
+
+    def _fold_scale(self, scales):
+        """Absorb the scale to the operator at output channel.
+
+        Args:
+            scales (dict): scales for smooth quant, {tensor_name: smooth quant scale}
+        """
+        remove_nodes = []
+        for node in self.model.nodes():
+            if node.op_type == "Mul"  and node.name.endswith("_smooth_mul") and node not in remove_nodes:
+                parent = self.model.get_parent(node, 0)
+                if parent is None:
+                    continue
+                if parent.op_type in self.could_absorb_optype and len(self.model.get_children(parent)) == 1:
+                    if node.output[0].split("_smooth_output")[0] in scales:
+                        if self.could_absorb_optype[parent.op_type](parent,
+                                1.0 / scales[node.output[0].split("_smooth_output")[0]]):
+                            remove_nodes.append(node)
+                            children = [i for i in self.model.nodes() if node.output[0] in i.input]
+                            for child in children:
+                                for idx, inp in enumerate(child.input):
+                                    if inp == node.output[0]:
+                                        child.input[idx] = node.input[0]
+        self.model.remove_nodes(remove_nodes)
+
+    def _dump_op_info(self, percentile, op_types, iterations, quantize_config=None):
+        """Dump op info for smooth quant.
+        
+        Args:
+            percentile (float): percentile of calibration to remove outliers
+            op_types (list): the op type to be smooth quantized
+            iterations (int): iterations
+            quantize_config (dict): quantize config
+        """
+        from neural_compressor.adaptor.ox_utils.calibration import ONNXRTAugment
+        augment = ONNXRTAugment(self.model,
+                                self.dataloader,
+                                [],
+                                iterations=list(range(0, iterations)),
+                                backend=self.backend,
+                                reduce_range=self.reduce_range)
+        self.max_vals_per_channel, self.shape_info, self.tensors_to_node = \
+            augment.calib_smooth(percentile, op_types, None)
+        for node in self.model.nodes():
+            for out in node.output:
+                if out in self.tensors_to_node and node.op_type in self.could_absorb_optype and \
+                    self.model.get_initializer(node.input[1]) is not None :
+                     self.ops_to_absorb.append(node.name)
+
+    def _get_output_loss(self, node_name, scale, calib_iter):
+        """Get output loss of specific node after inserting QDQ pair.
+        
+        Args:
+            node_name (str): node name
+            scale (float): scale of the specific node
+            calib_iter (int): iterations
+        """
+        from onnx import helper

```

Comment:
I am prone to code a bug here.  When quant_dequant input/activation of a node, the calibration input max should be adjusted using different alpha. Please double-check since I am not familiar with the code.

Reply:
@mengniwang95 please check the comments when you back, will merge the PR at first due to the code freeze.